### ISSUE TITLE
MPS max_pool2d_with_indices failure: destination values and indices length mismatch along axis

### ISSUE THREAD

**Issue by manuelcandales** [pytorch/pytorch#154882]
### 🐛 Describe the bug

max_pool2d_with_indices fails for some combinations of input shape and parameters.

```python
import torch

torch.ops.aten.max_pool2d_with_indices(
    torch.rand((1, 1, 4, 6), device="mps"),
    kernel_size=1,
    stride=2,
    padding=0,
    ceil_mode=True)
```

Error message:
```
/AppleInternal/Library/BuildRoots/01adf19d-fba1-11ef-a947-f2a857e00a32/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSNDArray/Kernels/MPSNDArrayPooling.mm:1980: failed assertion `Error: destination values and indices length mismatch along axis 0'
```

### Versions

torch==2.7.0

**Comment by jhavukainen** [comment]
Can confirm this is reproduces. Creating an MPSGraph unittest now to verify the problem is in the graph and not the way we call it.

**Comment by jhavukainen** [comment]
Using `ceilMode = True` with input shape `[1, 1, 4, 6]` and kernel size = `[1, 1]` with strides `[2, 2]` produces the output shape of `[1, 1, 3, 4]` according to the MPSGraph. IIUC the formula for the output dimensions should be 

`return ceil_mode ? ceil(input_dim / stride) : floor(input_dim / stride)`.

In this case for both dimensions the division resolves to an integer the output shape should be `[1, 1, 2, 3]` regardless of the ceil mode which would also solve the raised error of mismatch between the shape of values returned by MPSGraph and the container it tries to store the values into.

Filing an internal issue against the graph based on the misbehavior.

**Comment by jhavukainen** [comment]
Assigning to myself to figure out what the conditions are when MPSGraph miscalculates the shape and raise an error as the immediate patch up. 

Longer term I'll figure out if there's some limited set of conditions under which this fails which might let us massage the input shapes a little help graph do the right thing while waiting for them to really fix it. Or if not we'll probably need to add a separate kernel for this on PyTorch side.
